### PR TITLE
feat: different query/upsert fieldTypes

### PR DESCRIPTION
**Pull request by BrysonUrie**

### CODE REVIEW

File: src/table.ts [hunk 1]
```diff
@@ -99,7 +100,11 @@ export class Table<TableDef extends TableDefinition> {
     return newItem;
   }
 
-  private buildQBData(body: Partial<DocumentFromTable<TableDef>>) {
+  private buildQBData(
+    body: Partial<
+      QueryDocumentFromTable<TableDef> | UpsertDocumentFromTable<TableDef>
+    >,
+  ) {
```
Comment (Lil-Strudel) [inline src/table.ts]: Shouldnt this only take UpsertDocumentFromTable<TableDef>

**Review by Lil-Strudel — CHANGES_REQUESTED** [review]

File: src/table.ts (hunk 1 above)
Comment (BrysonUrie) [reply src/table.ts]: I dont think so.
The Update one function takes in a UpsertDocumentFromTable and uses the buildQBData function to create the data field in the request.

Comment (Lil-Strudel) [reply src/table.ts]: Im confused. Doesn't that just confirm what I said? The three places using buildQBData are:
updateOne
upsertMany
createOne

all of which use UpsertDocumentFromTable

Comment (BrysonUrie) [reply src/table.ts]: Total brain lapse, yes you are right. My bad

Comment (Lil-Strudel) [reply src/table.ts]: All g. Id rather have you say something instead of nothing

**Pull request merged**